### PR TITLE
Issue/1360 settings tracking

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
NA
#### The solution
Adding tracking to settings screen to find out how collectors use the app and which settings are useful
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java [hunk 1]
```diff
@@ -207,44 +222,67 @@ void onDownloadFormOptionTap() {
...
+                trackingHelper.logLanguageChanged(languages.get(position));
+            }
             presenter.saveAppLanguage(position, languages);
         }
     }
 
     @OnItemSelected(R.id.preference_image_size)
     void onImageSizeSelected(int position) {
-        presenter.saveImageSize(position);
+        if (trackChanges) {
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java]: Only here I see that the checking for whether tracking is enabled before logging an event. It isn't clear to me how that checking is handled in other places.

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java [hunk 2]
```diff
@@ -207,44 +222,67 @@ void onDownloadFormOptionTap() {
...
+            if (trackingHelper != null) {
+                trackingHelper.logImageSizeChanged(position);
+            }
+            presenter.saveImageSize(position);
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java]: Shouldn't this be outside of the `if (trackChanges) {}` statement?  i.e. it should happen whether tracking is on or not no?

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java (hunk 1 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java]: Oh this had nothing to do with tracking, I have renamed it

File: app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java (hunk 2 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java]: Oh this had nothing to do with tracking, I have renamed it

**Pull request merged**